File: Java/프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;
class Solution {
    public int solution(int[] citations) {
        Arrays.sort(citations);
        
        int hIndex = 0;
        int n = citations.length;
        
        for (int i = 0; i < n; i++) {
            // 해당 인덱스에서의 H-Index 조건 확인
            int h = n - i; 
            if (citations[i] >= h) {
                hIndex = h;
                break;
            }
        }
        
        return hIndex;
}
}

File: 프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    let h = 0
    const n = citations.length
    citations.sort((a,b) => a-b)
    
    for(let i = 0 ; i < n ; i++){
        for(let standard = 0 ; standard < 10001 ; standard++){
            if(citations[i] >= standard && n-i >= standard ){
                h = Math.max(h,standard)
            }else{
                break
            }
           
        }
    }
    
    return h
}